src/expr/src/vector_op/exp.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num_traits::Float;
+use risingwave_common::types::OrderedF64;
+
+use crate::{ExprError, Result};
+
+pub fn exp_f64(input: OrderedF64) -> Result<OrderedF64> {
+    let res = input.exp();
+    if res.is_infinite() {
+        Err(ExprError::NumericOutOfRange)
```

Comment:
FYI PG logic [here](https://github.com/postgres/postgres/blob/REL_15_2/src/backend/utils/adt/float.c#L1649) if we want to be fully compatible.

Reply:
let me track it in a new issue, I generally feel that If RW is more `powerful` than PG, it is ok 🤔 
Will correct it when users complain